components/automate-deployment/pkg/client/restore.go

```diff
@@ -295,38 +298,41 @@ func (r *DeploymentRestore) startDeploymentService(ctx context.Context) error {
 	}
 
 	var (
-		startTime  = time.Now()
-		timeout    = 20 * time.Second
 		sleepTime  = 500 * time.Millisecond
+		currentRel = "unknown"
 		desiredRel = deployment.ServiceFromManifest(r.a2Manifest, deploymentServiceName).Release()
+		doneChan   = make(chan struct{})
 	)
 
-	for {
-		// NOTE 2019-02-20: We assume that if the
-		// deployment-service is responding and returning the
-		// correct version than it is up. We don't currently
-		// depend on the Habitat health status here because
-		// that status can be delayed for up to 30 seconds if
-		// the first health check happens to fail.
-		currentRel, err := r.serviceRelease()
-		if err != nil {
-			currentRel = "unknown" // make the error message pretty
-		}
+	go func() {
```

Comment:
I'm not sure why we need a goroutine here. The call to will bail and you can check if the context is expired then?